### PR TITLE
Add Flash -> RAM copy

### PR DESCRIPTION
**Pull request by debuti**
By using cargo feature you can select to use RAM instead of XIP by copying the contents in flash to ram

### CODE REVIEW

**Review by thejpster — APPROVED** [review]
Looks OK to me.

**Comment by debuti** [comment]
Thanks, I tried but I cannot merge the changes as I have no write access

**Comment by jannic** [comment]
@debuti, do you have an example on how to use the copy-to-ram boot2?

I think I got it running with some wild tweaking of the link.x from cortex-m-rt, but then probe-rs doesn't like the result (`Unsupported address 0x20001314 for HW breakpoint. Breakpoint must be at address < 0x2000_0000.`), `elf2uf2` neither (`Detected RAM binary`, `Error: "Memory segment 0x10000000->0x10000100 is outside of valid address range for device"`).

After patching `elf2uf2` to force flash mode, a blinky example did work.

But this all seems much more complicated than the short description you added to README.md.

**Pull request merged**
